app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java

```diff
@@ -323,6 +326,70 @@ public static BgReading create(double raw_data, double filtered_data, Context co
                 }
 
                 bgReading.calculated_value = Math.min(400, Math.max(40, bgReading.calculated_value));
+                Log.w(TAG, "create: NEW VALUE CALCULATED AT: " + bgReading.calculated_value);
+
+                bgReading.save();
+                bgReading.perform_calculations();
+                Notifications.getInstance(context).notificationSetter(context);
+                BgSendQueue.addToQueue(bgReading, "create", context);
+            }
+        }
+        Log.w("BG GSON: ",bgReading.toS());
+
+        return bgReading;
+    }
+    public static BgReading create(double raw_data, Context context, Long timestamp) {
```

Comment:
Please note that alerts code looks at filtered data as well. If you can't
get the real value, need to set there the rawvalue.

Thanks
Tzachi

On Wed, May 20, 2015 at 12:26 AM, John <notifications@github.com> wrote:

> In app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java
> <https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/23#discussion_r30649200>
> :
>
> > @@ -323,6 +326,70 @@ public static BgReading create(double raw_data, double filtered_data, Context co
> >                  }
> >
> >                  bgReading.calculated_value = Math.min(400, Math.max(40, bgReading.calculated_value));
> > +                Log.w(TAG, "create: NEW VALUE CALCULATED AT: " + bgReading.calculated_value);
> > +
> > +                bgReading.save();
> > +                bgReading.perform_calculations();
> > +                Notifications.getInstance(context).notificationSetter(context);
> > +                BgSendQueue.addToQueue(bgReading, "create", context);
> > +            }
> > +        }
> > +        Log.w("BG GSON: ",bgReading.toS());
> > +
> > +        return bgReading;
> > +    }
> > +    public static BgReading create(double raw_data, Context context, Long timestamp) {
>
> Looking at the code here, I would prefer to keep the create(double
> raw_data, double filtered_data, context, timestamp one, and have the
> create(raw_data, context, timestamp) one call it with raw_data sent as both
> itself and filtered data. If I use the other, filtered data is not stored,
> raw data is stored in the filtered_data field/attribute. John Stevens M - +61
> 418526349
> … <#14d6e11d39b21c10_>
>  On 20 May 2015, at 07:00, tzachi-dar <notifications@github.com> wrote:
> In app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java: > @@
> -323,6 +326,70 @@ public static BgReading create(double raw_data, double
> filtered_data, Context co > } > > bgReading.calculated_value =
> Math.min(400, Math.max(40, bgReading.calculated_value)); > + Log.w(TAG,
> "create: NEW VALUE CALCULATED AT: " + bgReading.calculated_value); > + > +
> bgReading.save(); > + bgReading.perform_calculations(); > +
> Notifications.getInstance(context).notificationSetter(context); > +
> BgSendQueue.addToQueue(bgReading, "create", context); > + } > + } > +
> Log.w("BG GSON: ",bgReading.toS()); > + > + return bgReading; > + } > +
> public static BgReading create(double raw_data, Context context, Long
> timestamp) { If we have a function that is never called we should remove
> it... … — Reply to this email directly or view it on GitHub.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/23/files#r30649200>
> .
>